Lib/shutil.py

```diff
@@ -529,9 +529,11 @@ def _copytree(entries, src, dst, symlinks, ignore, copy_function,
             else:
                 # Will raise a SpecialFileError for unsupported file types
                 copy_function(srcobj, dstname)
-        # catch the Error from the recursive copytree so that we can
-        # continue with other files
+        except SameFileError as err:
```

Comment:
I think you need the same handler for `SpecialFileError`, which is also instantiated with a string.

Reply:
(and possibly the other errors too, I haven't attempted to trace them all).

Using `ExceptionGroup` here might make things cleaner, but is probably backwards-incompatible.

Reply:
I will try to reproduce that with a test.

Reply:
I could not make a scenario that would yield this error (a splitted string in the error message). There is already a test function that checks the error string when a SpecialFileError is provoked with a pipe 

https://github.com/python/cpython/blob/ffece5590e95e89d3b5b6847e3beb443ff65c3db/Lib/test/test_shutil.py#L999

which already passes and that behaviour is not changed in this PR. In that case there is recursion involved, and then the existing code works. The bug with splitted strings only occurs when there is no recursion.